Connection.js

```diff
@@ -417,11 +417,25 @@ Connection.prototype.answerHandshake = function (lines) {
 	sha1 = crypto.createHash('sha1')
 	sha1.end(this.key + '258EAFA5-E914-47DA-95CA-C5AB0DC85B11')
 	key = sha1.read().toString('base64')
-	this.socket.write(this.buildRequest('HTTP/1.1 101 Switching Protocols', {
+
+	headers2send = {
 		Upgrade: 'websocket',
 		Connection: 'Upgrade',
 		'Sec-WebSocket-Accept': key
-	}))
+	}
+
+	// subprotocols support
+	// https://tools.ietf.org/html/rfc6455#page-7
+	if( 'sec-websocket-protocol' in this.headers ){
+		protocol = this.headers['sec-websocket-protocol'];
+		this.protocols = protocol.split(',').map(function(v){return v.trim()});
+		headers2send['Sec-WebSocket-Protocol'] = protocol;
```

Comment:
From the spec:
> The server selects one or none of the acceptable protocols and echoes
   that value in its handshake to indicate that it has selected that
   protocol.

The client may send multiple protocols, but the server should pick one of them. The way the code above is written, it would wrongly echo back the whole list.